chapters/annotations.tex

```diff
@@ -458,6 +459,68 @@ \section{Single Use of Class}\label{annotation-for-single-use-of-class}\label{si
 The annotation \fmtannotationindex{singleInstance} in a class indicates that there should only be one component instance of the class, and it should be in the same scope as the class is defined.
 The intent is to remove the class when the component is removed and to prevent duplication of the component.
 
+\subsection{Connection Restrictions}\label{connection-restrictions}
+A connector component declaration may have the following annotation:
+\begin{lstlisting}[language=modelica]
+annotation(mustBeConnected = "message");
+\end{lstlisting}%
+\annotationindex{mustBeConnected}
+
+It makes it an error if the connector is not connected from the outside (for a conditional connector this check is only active if the connector is enabled).
+For an array of connectors it applies separately to each element.
+
+\begin{nonnormative}
+This annotation is intended for non-causal connectors, see \cref{restrictions-of-connections-and-connectors}.
+It is particularly suited for stream connectors, see \cref{stream-connectors}.
+\end{nonnormative}
+
+\begin{example}
+This can be used for some optional connectors that should be connected when conditionally enabled.
+\begin{lstlisting}[language=modelica]
+partial model PartialWithSupport
+  Flange_a flange_a;
+  Flange_b flange_b;
+  Support support($\ldots$)
+    if useSupport
+    annotation(
+      mustBeConnected = "Support connector should be connected if activated.");
+  $\ldots$
+end PartialWithSupport;
+\end{lstlisting}
+\end{example}
+
+A connector component declaration may have the following annotation:
+\begin{lstlisting}[language=modelica]
+annotation(mayOnlyConnectOnce = "message");
+\end{lstlisting}%
+\annotationindex{mayOnlyConnectOnce}
+
+It makes it an error if the connector is connected from the outside and:
+\begin{itemize}
+\item For non-stream connectors the connection set has more than two elements.
+\item For stream connectors (see \cref{stream-connectors}), the connection set has more than two elements whose flow variable may be negative (based on the \lstinline!min!-attribute).
```

Comment:
Is the error expected to trigger during translation, or is it expected that changing the `min` parameter at initialization time could resolve the situation?  If not (and short of `min` being a constant attribute), one could consider something more aggressive, like this:
```suggestion
\item For stream connectors (see \cref{stream-connectors}), the connection set has more than two elements whose flow variable may be negative (based on evaluation of the \lstinline!min!-attribute).
```